sending.go

```diff
@@ -30,15 +33,62 @@ func (pinger *Pinger) PingAttempts(remote *net.IPAddr, timeout time.Duration, at
 // Ping sends a single Echo Request and waits for an answer. It returns
 // the round trip time (RTT) if a reply is received in time.
 func (pinger *Pinger) Ping(remote *net.IPAddr, timeout time.Duration) (time.Duration, error) {
-	if timeout <= 0 {
-		return 0, errors.New("zero timeout")
+	ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(timeout))
+	defer cancel()
+	return pinger.PingContext(ctx, remote)
+}
+
+// PingContext sends a single Echo Request and waits for an answer. It returns
+// the round trip time (RTT) if a reply is received before cancellation of the context.
+func (pinger *Pinger) PingContext(ctx context.Context, remote *net.IPAddr) (time.Duration, error) {
+	req := simpleRequest{}
+	seq, err := pinger.sendRequest(remote, &req)
+
+	// wait for answer
+	select {
+	case <-req.wait:
+		// already dequeued
+		err = req.result
+	case <-ctx.Done():
+		// dequeue request
+		pinger.removeRequest(seq)
+
+		err = &timeoutError{}
 	}
 
-	seq := uint16(atomic.AddUint32(&sequence, 1))
-	req := request{
-		wait: make(chan struct{}),
+	if err != nil {
+		return 0, err
+	}
+	return req.roundTripTime()
+}
+
+// PingMulticast sends a single Echo Request and waits a given time for the answer(s)
```

Comment:
Could you explain the channel semantics in the documentation here? I.e., the channel will be closed when the context is cancelled (I think?).

Also, maybe add an example to the documentation?
```go
ctx, canc := context.WithTimeout(context.Background(), 1 * time.Second)
defer canc()
responses, err := PingMulticast(ctx, net.ParseIP("ff02::1%eth0"))
if err != nil {
  return err
}
for resp := range responses {
  fmt.Println(resp)
}
```